### PR TITLE
Wip feature/add cli version feedback

### PR DESCRIPTION
**Pull request by adam2k**
This is meant for a preview of the fix for:

Fix https://github.com/astronomer/issues/issues/923

### CODE REVIEW

File: cmd/airflow.go [hunk 1]
```diff
@@ -45,7 +45,10 @@ func newAirflowRootCmd(client *houston.Client, out io.Writer) *cobra.Command {
 		Aliases:    []string{"a"},
 		Short:      "Manage airflow projects",
 		Long:       "Airflow projects are a single top-level directory which represents a single production Airflow deployment",
-		Deprecated: "could please use new command instead `astro dev [subcommands] [flags]`",
+		Deprecated: "Please use new command instead `astro dev [subcommands] [flags]`",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			PersistentPreRunCheck(client, cmd, out)
```
Comment (andriisoldatenko) [inline cmd/airflow.go:50]: let rename to something like: checkVersion or something more readable.

File: cmd/prerun.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/astronomer/astro-cli/houston"
+	"github.com/astronomer/astro-cli/messages"
+	"github.com/astronomer/astro-cli/version"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+// PersistentPreRunCheck for validation of the CLI vs Deployment Version
+func PersistentPreRunCheck(client *houston.Client, cmd *cobra.Command, out io.Writer) {
+	ac := deployment.GetAppConfig()
```
Comment (andriisoldatenko) [inline cmd/prerun.go:19]:
```suggestion
	appCfg := deployment.GetAppConfig()
```

File: cmd/prerun.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	} else if isAheadMajor(cv, dv) {
+		color.Yellow(messages.WARNING_DOWNGRADE_VERSION, cv, dv)
+	}
+}
+
+func isBehindMajor(cv string, dv string) bool {
+	fm := formatMajor(dv)
+	fc := formatLtConstraint(fm)
+	m := getConstraint(fc)
+	v := getVersion(cv)
```
Comment (andriisoldatenko) [inline cmd/prerun.go:46]: general note, try to avoid `m`, `v` etc. better short but concise

Comment (adam2k) [reply cmd/prerun.go:46]: I can change to `majCon` and `ver`?

**Pull request merged**